dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,112 @@
+[[getting-traffic-into-cluster]]
+= Getting Traffic Into The Cluster
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- When possible use the Router,
+- otherwise for TCP/UDP configure the service to provide access.
+
+The admin configures DNS records to point to IP addresses in the cluster
+that are exposed outside the cluster.  These addresses are commonly the
+IP addresses of nodes, Virtual IP addresses (VRRP) or Service NodePort
+addreses, Service externalIPs, Service ingressIP, etc..  The admin must
+make sure the routing tables permit the request to reach the IP address
+and iptables accept the packets as well.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Using a router] to serve HTTP/HTTPS(SNI)/TLS (SNI)
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#using-tcp[Service NodePort] and VIP
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#pod-network-to-outside[Access pod network with a load balancer]
+
+You can also use the wildcard DNS for the router + the nodeport
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A router is configured
+to accept external reuqests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+Setup router nodes (either TCP load balance or ipfailover VRRP setup) in
+an HA config.  When asking for nodeports, use the router nodes with their
+VIPs (VRRP) or with a range TCP load balancer (cloud, F5, ingerssIP) as
+the destination IP.
+
+Since the router listens on ports 80 and 443 the number of routers is
```

Comment:
OK I will work that in. I am revising the doc in general based on your 
comments. New draft shortly...
phil


On 09/12/2016 01:10 PM, Ben Bennett wrote:
>
> In dev_guide/getting_traffic_into_cluster.adoc 
> <https://github.com/openshift/openshift-docs/pull/2827#discussion_r78413098>:
>
> > +endif::[]
> > +
> > +
> > +[[using-a-router]]
> > +== Using a Router
> > +
> > +This is the most common way to access the cluster. A router is configured
> > +to accept external reuqests and proxy them based on the configured routes.
> > +This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
> > +
> > +Setup router nodes (either TCP load balance or ipfailover VRRP setup) in
> > +an HA config.  When asking for nodeports, use the router nodes with their
> > +VIPs (VRRP) or with a range TCP load balancer (cloud, F5, ingerssIP) as
> > +the destination IP.
> > +
> > +Since the router listens on ports 80 and 443 the number of routers is
>
> "Think about it this way" is fine. But if you are going to give it as 
> a categorical restriction, then you must at least modulate it, or just 
> ditch the section. The only real reason to run multiple routers would 
> be for sharding, and we should mention that as a benefit with a link. 
> And I think the sharding section talks about how to work around the 
> port allocation problem.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openshift/openshift-docs/pull/2827/files/b1aadf12f94f3982da1f9b8aca31982cd8fe4b86#r78413098>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ANUgerJ_6zLR92HKrPcmE1l6HqQIXsvJks5qpYefgaJpZM4J6qBl>.
>